clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -354,6 +355,476 @@ fatbinary(ArrayRef<std::pair<StringRef, StringRef>> InputFiles,
 }
 } // namespace amdgcn
 
+namespace sycl {
+// This utility function is used to gather all SYCL device library files that
+// will be linked with input device files.
+// The list of files and its location are passed from driver.
+static Error getSYCLDeviceLibs(SmallVector<std::string, 16> &DeviceLibFiles,
+                               const ArgList &Args) {
+  StringRef SYCLDeviceLibLoc;
+  if (Arg *A = Args.getLastArg(OPT_sycl_device_library_location_EQ))
+    SYCLDeviceLibLoc = A->getValue();
+  else
+    return createStringError(inconvertibleErrorCode(),
+                             "SYCL device library location is invalid.");
+  if (Arg *A = Args.getLastArg(OPT_sycl_device_lib_EQ)) {
+    if (A->getValues().size() == 0)
+      return createStringError(
+          inconvertibleErrorCode(),
+          "Number of device library files cannot be zero.");
+    for (StringRef Val : A->getValues()) {
+      SmallString<128> LibName(SYCLDeviceLibLoc);
+      llvm::sys::path::append(LibName, Val);
+      if (llvm::sys::fs::exists(LibName))
+        DeviceLibFiles.push_back(std::string(LibName));
+      else
+        return createStringError(inconvertibleErrorCode(),
+                                 std::string(LibName) +
+                                     " SYCL device library file is not found.");
+    }
+  }
+  return Error::success();
+}
+
+static bool isStaticArchiveFile(const StringRef Filename) {
+  if (!llvm::sys::path::has_extension(Filename))
+    // Any file with no extension should not be considered an Archive.
+    return false;
+  llvm::file_magic Magic;
+  llvm::identify_magic(Filename, Magic);
+  // Only .lib and archive files are to be considered.
```

Comment:
```suggestion
  // Only archive files are to be considered.
```
Maybe you will be adding a `.lib` check in the future?